src/Text/Toml/Parser.hs

```diff
@@ -5,5 +5,21 @@ module Text.Toml.Parser
     ( parseToml
     ) where
 
-parseToml = undefined
+import Text.Toml.Tokenizer
+import Text.Toml.Types.Tokens
+import Text.Toml.Types.Toml
 
+import Control.Monad ((<=<))
+import Data.Text (Text)
+import Data.Monoid ((<>))
+
+parseToml :: Text -> Either String Toml
+parseToml = interpret empty <=< tokenize
+
+interpret :: Toml -> [Token] -> Either String Toml
+interpret toml [] = Right toml
+interpret toml (IdentifierT _ n:EqualT _:KeywordTrueT _:rest) =
+    interpret (insert n (TBoolean True) toml) rest
+interpret toml (IdentifierT _ n:EqualT _:KeywordFalseT _:rest) =
+    interpret (insert n (TBoolean False) toml) rest
+interpret _ t = Left $ "Unexpected token: " <> show t
```

Comment:
Ah! I think I thought that file was part of the query side and didn't look at it. I actually just spent about twenty minutes re-implementing some of it :) I'll update this PR when I have the same bool test passing with that approach, that'll be much nicer.